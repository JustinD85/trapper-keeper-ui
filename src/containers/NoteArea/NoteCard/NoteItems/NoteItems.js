import React from "react";
import PropTypes from "prop-types";

const NoteItems = ({ noteItems }) => (
  <div>
    {noteItems.map(item => (
      <div key={item.id} id={item.id}>
        {item.text}
      </div>
    ))}
  </div>
);

export default NoteItems;

NoteItems.propTypes = {
  noteItems: PropTypes.array
};
