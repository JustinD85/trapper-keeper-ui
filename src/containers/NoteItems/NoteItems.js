import React from 'react';
import PropTypes from 'prop-types';
import { Divider, Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles';

const styles = {
  divider: {
    backgroundColor: '#ff8922'
  },
  strikethrough: {
    textDecoration: 'line-through'
  }
};

export const NoteItems = ({ noteItems, classes }) => {
  const jsxItems = noteItems.map(item => (
    <div iscompleted={JSON.stringify(item.isCompleted)} key={item.id} >
      <Typography 
        id={item.id} 
        className={item.isCompleted ? classes.strikethrough : ''} 
        style={{color: '#48494a'}}
        variant='body2'>
        {item.text}
      </Typography>
      <Divider light/>
    </div>
  ))
  return (
  <div>
    {jsxItems.filter(item => (
      item.props.iscompleted === 'false'
    ))}
    <Divider className={classes.divider}/>
    {jsxItems.filter(item => (
      item.props.iscompleted === 'true'
    ))}
  </div>
  )
};

NoteItems.propTypes = {
  noteItems: PropTypes.array
};

export default withStyles(styles)(NoteItems)