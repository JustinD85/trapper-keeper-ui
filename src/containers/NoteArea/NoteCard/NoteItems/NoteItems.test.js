import React from 'react';
import NoteItems from './NoteItems';
import { shallow } from 'enzyme';

describe('NoteItems', () => {
  let wrapper;
  let mockItems = [{text: 'Dylan', id: 1}, 
  {text: 'Justin', id: 2},
  {text: 'Shannon', id: 3}]

  beforeEach(() => {
    wrapper = shallow(<NoteItems noteItems={mockItems}/>)
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});