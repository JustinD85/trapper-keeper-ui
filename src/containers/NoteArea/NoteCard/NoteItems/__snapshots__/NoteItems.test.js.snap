// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItems should match the snapshot 1`] = `
<div>
  <div
    id={1}
    key="1"
  >
    Dylan
  </div>
  <div
    id={2}
    key="2"
  >
    Justin
  </div>
  <div
    id={3}
    key="3"
  >
    Shannon
  </div>
</div>
`;
